source/develop/release-management/features/network/lldp.html.md

```diff
@@ -0,0 +1,448 @@
+      name: type
+      type: string
+      datatype: uint
+
+    - description: organizationally unique identifier
+      name: oui
+      type: string
+      datatype: uint
+
+    - description: organizationally defined subtype
+      name: subtype
+      type: string
+      datatype: uint
+
+
+Host.getNicNeighbor:
+  added: '4.2'
+  description: Get information about the neighbor of a NIC
+  params:
+    defaultvalue: no-default
+    description: Device name to query for neigbor
+    name: devicename
+    type: string
+  return:
+    description: Information about the neighbor of a NIC
+    type:
+    - *Tlv
+```
+
+To implement the [follow-up features](#follow-up-features), the LLDP information
+would be required cyclic and `VdsmCapabilities` might be extended in future.
+
+## Entity Description
+
+The initial LLDP information is a list of type, length, value (TLV) tuples.
+To express multiple values, e.g. to list the allowed VLANs on a switch port,
+the TLV is sent multiple times.
+The value may contain structured data.
+This way the LLDP data is non-uniform.
+This non-uniform LLDP data will be mapped to the following uniform entities.
+
+The entity `Tlv` represents the TLV received physically by the oVirt host's NIC.
+A `Tlv` is associated to a `VdsNetworkInterface`, called
+[`HostNic` in REST-API](http://ovirt.github.io/ovirt-engine-api-model/4.2/#types/host_nic),
+has a `name`, a `type`, a list of `properties` and optional an `oui` and a
+`subtype`.
+
+![Class diagramm of Tlv](../../../../images/wiki/EntityTlv.png  "Class diagramm of Tlv")
+
+The `name` is a human readable string to describe what the value is about and
+may not be unique. The name is redundant, because it could be created from
+`type` and the optional properties "oui" and "subtype". The purpose of `name` is
+to simplify the reading of the TLV.
+The `name` of a property is exactly the same string which is used in
+[IEEE 802.1AB][1] chapter 8.
+
+The `type` is the TLV type of [IEEE 802.1AB][1].
+
+Organizationally Specific `Tlv`s have the `type` of `0x7f` and the attributes
+`oui` and `subtype`.
+The `oui` holds the organizationally unique identifier, and the `subtype` the
+organizationally defined subtype.
+
+The `properties` are used to represent structured information transported by the
+TLV.
+
+## CRUD
+
+For REST-API only not required.
+
+## User Experience
+
+### REST-API
+
+The LLDP information are read-only. They could be read by a request like
+
+```
+GET https://ovirtexampleengine/ovirt-engine/api/hosts/bdb7c9f9-e10d-4af0-af9c-a7f0c7235fdd/nics/0ac967ac-af2c-4628-997d-ef845edeeeb0/neighbor
+```
+
+The response in XML for the [SampleCaptures from Wireshark's wiki][5] will look like this
+
+```xml
+<host_nic_neighbor>
+  <host_nic href="/ovirt-Engine/api/hosts/bdb7c9f9-e10d-4af0-af9c-a7f0c7235fdd/nics/0ac967ac-af2c-4628-997d-ef845edeeeb0" id="0ac967ac-af2c-4628-997d-ef845edeeeb0"/>
+  <tlvs>
+    <tlv>
+      <name>Chassis ID</name>
```

Comment:
Just trying to understand if this is possible or makes sense (not sure if it can be used on our rest):
As the name is just a helper that describes what the type is, can;t we use it as a property of the type element?
Like: <type name="Chassis ID">0x1</type>